CRM/Member/Form/Membership.php

```diff
@@ -1259,8 +1259,12 @@ public function submit(): void {
     $contributionId = $this->ids['Contribution'] ?? CRM_Member_BAO_Membership::getMembershipContributionId($this->getMembershipID());
     $membershipIds = $this->_membershipIDs;
     if ($this->getSubmittedValue('send_receipt') && $contributionId && !empty($membershipIds)) {
-      $contributionDetails = CRM_Contribute_BAO_Contribution::getContributionDetails(CRM_Export_Form_Select::MEMBER_EXPORT, $this->_membershipIDs);
-      if ($contributionDetails[$this->getMembershipID()]['contribution_status'] === 'Completed') {
+      $contributionStatus = \Civi\Api4\Contribution::get()
```

Comment:
It looks like the "before" code was not checking permissions and the "after" code is checking permissions.

Reply:
Hmm - that's probably OK - since I think that form would probably rely on the editor being able to access the contribution